scanners/sslyze.py

```diff
@@ -250,11 +251,11 @@ def parse_sslyze(raw_json):
         else:
             data['certs']['key_length'] = None
 
-        if leaf_key.__class__ == cryptography.hazmat.backends.openssl.rsa._RSAPublicKey:
+        if isinstance(leaf_key, rsa.RSAPublicKey):
             leaf_key_type = "RSA"
-        elif leaf_key.__class__ == cryptography.hazmat.backends.openssl.dsa._DSAPublicKey:
+        elif isinstance(leaf_key, dsa.DSAPublicKey):
             leaf_key_type = "DSA"
-        elif leaf_key.__class__ == cryptography.hazmat.backends.openssl.ec._EllipticCurvePublicKey:
+        elif isinstance(leaf_key, ec.EllipticCurvePublicKey):
```

Comment:
This change wasn't related to multiple gatherers and probably should be in its own PR, but it also doesn't change behavior -- it's addressing the advice in https://github.com/pyca/cryptography/issues/3541#issuecomment-301367551 as to how to properly introspect the key type of a certificate object returned by the Python `cryptography` module.